### PR TITLE
Add PropTypes to all of our React components

### PR DESCRIPTION
**Pull request by SirCmpwn**
This brings our linter errors down to zero - so this PR also turns on linting during the Travis build and will fail builds for introducing linter errors.

### CODE REVIEW

File: src/components/Dropdown.js [hunk 1]
```diff
@@ -27,3 +27,10 @@ export default function Dropdown(props) {
     </span>
   );
 }
+
+Dropdown.propTypes = {
```
Comment (eatonphil) [inline src/components/Dropdown.js]: I thought you could also specify propTypes within the class definition. Does the linter not allow that or are we just going with this style?

Comment (SirCmpwn) [inline src/components/Dropdown.js]: That's not possible if you're doing ES6 classes instead of using `React.createClass`.

Comment (eatonphil) [inline src/components/Dropdown.js]: Should be possible https://toddmotto.com/react-create-class-versus-component/#proptypes-and-getdefaultprops

Comment (SirCmpwn) [inline src/components/Dropdown.js]: This link demonstrates my point.

Comment (eatonphil) [inline src/components/Dropdown.js]: Sorry, wasn't thinking.

**Comment by eatonphil** [comment]
Tests not passing?

File: src/components/Dropdown.js (hunk 1 above)
Comment (SirCmpwn) [inline src/components/Dropdown.js]: np

**Comment by SirCmpwn** [comment]
Pushed a fix that should deal with tests.

**Comment by eatonphil** [comment]
Wow these really take a while.

**Comment by SirCmpwn** [comment]
Tests passing.

**Comment by eatonphil** [comment]
Cool. ready to merge?

**Comment by SirCmpwn** [comment]
Yes.

**Pull request merged**